platforms/core-runtime/base-services/src/test/groovy/org/gradle/internal/service/ServiceRegistryBuilderTest.groovy

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright 2023 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.internal.service
+
+import org.gradle.internal.service.scopes.Scopes
+import spock.lang.Specification
+
+class ServiceRegistryBuilderTest extends Specification {
+
+    def "creates a scoped service registry when setting a scope"() {
```

Comment:
💭 It might be better to test the functionality rather than the type here as in the integration test.  You could try to register Mocked services with the correct and incorrect scope and make sure they are accepted and rejected appropriately.

Reply:
Good idea, I updated the test